File: backend/TaskManagementSystem/TaskSystem/Models/TaskModel.cs
using TaskSystem.Models;
using TaskSystem.Models.Enum;

namespace TaskSystem.Models;

public class TaskModel
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public bool IsCompleted { get; set; }
    public DateTime DueDate { get; set; }
    public Priority Priority { get; set; }
    public string UserId { get; set; }
    public AppUser User { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

public class TaskDTO
{
    public int Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public bool IsCompleted { get; set; }
    public DateTime DueDate { get; set; }
    public Priority Priority { get; set; }
    public string UserId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

public class CreateTaskDTO
{
    public string Title { get; set; }
    public string Description { get; set; }
    public DateTime DueDate { get; set; }
    public Priority Priority { get; set; }
}

public class UpdateTaskDTO
{
    public string Title { get; set; }
    public string Description { get; set; }
    public DateTime DueDate { get; set; }
    public Priority Priority { get; set; }
}

public class TaskFilterParams
{
    public bool? IsCompleted { get; set; }
    public Priority? Priority { get; set; }
    public DateTime? DueDateFrom { get; set; }
    public DateTime? DueDateTo { get; set; }
    public string? SearchTerm { get; set; }
    public string? SortBy { get; set; }
    public bool SortDescending { get; set; }
}

File: frontend/src/models/TaskModel.ts

import { Priority } from './Priority';

export interface TaskModel {
  id: number;
  title: string;
  description: string;
  isCompleted: boolean;
  dueDate: Date;
  priority: Priority;
  userId: string;
  createdAt: Date;
  updatedAt?: Date;
}

export interface TaskDTO {
    id: number;
    title: string;
    description: string;
    isCompleted: boolean;
    dueDate: Date;
    priority: Priority;
    userId: string;
    createdAt: Date;
    updatedAt?: Date;
  }
  export interface CreateTaskDTO {
    title: string;
    description: string;
    dueDate: Date;
    createdAt: Date;
    priority: Priority;
  }
  export interface UpdateTaskDTO {
    title: string;
    description: string;
    dueDate: Date;
    priority: Priority;
  }

  export interface TaskFilterParams {
    isCompleted?: boolean;
    priority?: Priority;
    dueDateFrom?: Date;
    dueDateTo?: Date;
    searchTerm?: string;
    sortBy?: string;
    sortDescending: boolean;
  }